projects/step-frontend/src/lib/modules/_common/components/sidebar/sidebar.component.ts

```diff
@@ -21,19 +23,24 @@ import {
 } from '@exense/step-core';
 import { ILocationService } from 'angular';
 import { VersionsDialogComponent } from '../versions-dialog/versions-dialog.component';
+import { MENU_ITEMS } from '../../injectables/menu-items';
+import { Subject, takeUntil } from 'rxjs';
 
 @Component({
   selector: 'step-sidebar',
   templateUrl: './sidebar.component.html',
   styleUrls: ['./sidebar.component.scss'],
   encapsulation: ViewEncapsulation.None,
 })
-export class SidebarComponent implements AfterViewInit, OnDestroy {
+export class SidebarComponent implements OnInit, OnDestroy {
   @ViewChildren('mainMenuCheckBox') mainMenuCheckBoxes?: QueryList<ElementRef>;
 
   sideBarOpen: boolean = true;
-
+  private terminator$ = new Subject<void>();
   private locationStateSubscription: any;
+  readonly _menuItems$ = inject(MENU_ITEMS).pipe(takeUntil(this.terminator$));
```

Comment:
why is this prefixed with an underscore? and why is inject used instead of the injection to happen in the constructor? I think I actually tried this at some point and it was not even working, but even if it does work, I would say not to apply it until a migration to a newer Angular version has begun

Reply:
@bojidaryovchev `inject` is available since v14. The project has been moved to v14 since the end of November